09-intro-to-bcdata.Rmd

```diff
@@ -17,9 +17,133 @@ knitr_fig_path("09-")
 ```
 
 
-# Introduction to bcdata package
+# Introduction to the bcdata package
+
+The bcdata R package contains functions for searching & retrieving data directly from the B.C. Data Catalogue.
 
 ```{r}
 library(bcdata)
 ```
 
+Let's look in the B.C. Data Catalogue for some data on libraries in British Columbia:
+
+```{r}
+bcdc_search("public libraries")
+```
+
+There are many catalogue records with data relating to libraries. Let's examine the first returned record more closely:
+
+
+```{r}
+bcdc_get_record("bc-public-libraries-systems-branches-and-locations")
+```
+
+Next, we can use the `bcdc_get_data` function to load this data directly into R:
+
+```{r}
+bcdc_get_data(record = "bc-public-libraries-systems-branches-and-locations")
+
+lib_raw <- bcdc_get_data(record = 'bc-public-libraries-systems-branches-and-locations',
+                          resource = 'ed17f111-fb39-46b3-89aa-5375592fbb01')
+```
+
+
+> ## Challenge 1 (5 minutes)
+>
+> Take a few minutes to search the B.C. Data Catalogue for data sets for a different topic. Identify and import one tabular data set into your R session > (e.g. a CSV or XLS resource).
+>
+
+
+Let's return to our libraries data. As we learned earlier in the workshop, a useful first step is to get familiar with the data.
+
+```{r}
+str(lib_raw)
+names(lib_raw)
+```
+
+Let's tidy up this data frame and find out how many books&mdash;physical materials or volumes&mdash;are available within each City:
+
+```{r}
+library(dplyr)
+
+lib_tidy <- lib_raw %>%
+  select(`Library System`, Location, City, num_books = "460x Total branch physical materials, volumes held")
+
+
+sum_books <- lib_tidy %>% 
+  group_by(City, `Library System`) %>% 
+  summarize(total_books = sum(num_books))
+```
+
+> ## Challenge 2 (10 minutes)
+>
+> Using `dplyr`, determine the total number of books within each Library System in British Columbia.
+>
+>
+>```{r}
```

Comment:
It might be worth formatting the answer the same as other places in the book a la: https://github.com/bcgov/ds-intro-to-r-2-day/blob/30d029c77830744321faba567c512a1555de4727/08-tidyr.Rmd#L88-L97

Reply:
Done.